flybywiresim/a32nx
# refactor(tcas): fixed several issues with TCAS and traffic display

Fixes #[issue_no]

## Summary of Changes

Fixed several edge cases with traffic display and TCAS computer
- More traffic should be shown above ~5000 feet as an issue with the ground level being incorrectly calculated has been fixed.
- Traffic Advisories should now persist for at least 4 seconds before being dismissed, before the expiry was not being respected.
- In some cases, the ND would fail to correctly display TA/RA symbols for the relevant intruders, this should now always work.

## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub): 2Cas#1022

## Testing instructions


Test with both injected traffic (VATSIM/FSLTL) and live traffic (MSFS Live Traffic, not AI Traffic)

1. Load in C&D at FAOR (or another high altitude airfield), with either multiplayer, live traffic or VATSIM/FSLTL traffic enabled (live traffic and traffic from addons i.e. FSLTL preferred)
2. Power on, Turn on ATC XPDR and switch TCAS mode to TA/RA, etc.
3. Check that no traffic that is parked at a gate or taxiing on ground is displayed on the traffic display.
4. Load in on approach at FAOR (or another high altitude airfield) and check that no traffic that is on ground is displayed on the ND. 
5. Fly up to >10000 ft, look for and try and intercept local traffic (this may be harder to do with injected traffic. Check that traffic that is in range (-27000 ft to 27000 ft on all TCAS settings (ABV/BEL/ALL), extending to 99000 ft in either direction (ABV/BLW setting) is displayed. Note that the maximum number of displayed intruders is 8, if there is more 8 intruders, the furthest/non-intersecting traffic will not be displayed.
6. Attempt to trigger TAs and RAs
7. Disobey RA instructions at first, then obey the direction, observe that RA dismisses correctly when the conflict is cleared. 


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
